robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowPendingIntent.java

```diff
@@ -134,37 +186,30 @@ public int getFlags() {
     return flags;
   }
 
+  public boolean isCanceled() {
+    return canceled;
+  }
+
+  @Implementation
+  public String getTargetPackage() {
+    return getCreatorPackage();
+  }
+
+  @Implementation(minSdk = 17)
+  public String getCreatorPackage() {
+    return (creatorPackage == null)
+        ? savedContext.getPackageName()
+        : creatorPackage;
+  }
+
+  public void setCreatorPackage(String creatorPackage) {
+    this.creatorPackage = creatorPackage;
+  }
+
   @Override
   @Implementation
   public boolean equals(Object o) {
-    if (this == o) return true;
```

Comment:
How so? This is actually quite similar to the real PendingIntent equals, which compares the out of process binder target objects for equality (since the reference truth for PendingIntents is stored out of process). In a robolectric context PendingIntents will always be in process so we should be able to use a direct reference equality comparison.

Obviously this isn't passing the Travis tests though :/ This passed locally, running the tests in a different context than Travis, so I'm struggling to understand why there's a difference. Will dig further into this today.